torchvision/csrc/cpu/ROIAlign_cpu.cpp

```diff
@@ -134,18 +135,23 @@ void ROIAlignForward(
     int roi_batch_ind = offset_rois[0];
 
     // Do not using rounding; this implementation detail is critical
-    T roi_start_w = offset_rois[1] * spatial_scale;
-    T roi_start_h = offset_rois[2] * spatial_scale;
-    T roi_end_w = offset_rois[3] * spatial_scale;
-    T roi_end_h = offset_rois[4] * spatial_scale;
-    // T roi_start_w = round(offset_rois[0] * spatial_scale);
-    // T roi_start_h = round(offset_rois[1] * spatial_scale);
-    // T roi_end_w = round(offset_rois[2] * spatial_scale);
-    // T roi_end_h = round(offset_rois[3] * spatial_scale);
+    T offset = aligned ? (T)0.5 : (T)0.0;
+    T roi_start_w = offset_rois[1] * spatial_scale - offset;
+    T roi_start_h = offset_rois[2] * spatial_scale - offset;
+    T roi_end_w = offset_rois[3] * spatial_scale - offset;
+    T roi_end_h = offset_rois[4] * spatial_scale - offset;
 
     // Force malformed ROIs to be 1x1
     T roi_width = std::max(roi_end_w - roi_start_w, (T)1.);
     T roi_height = std::max(roi_end_h - roi_start_h, (T)1.);
```

Comment:
@ppwwyyxx 
In detecton2, Only RoIAlign Cpu version raise error about negative size.
But the cuda version just clamping `1.0`(using max)

Which do you think is the better way to raise an error or use max?

Reply:
It's mainly because it's harder to raise error from a cuda kernel. Maybe there is a nice way I'm not aware of.

Reply:
We could have an assert, but this makes the current context invalid (needs to reboot the Python kernel), which is what we do in PyTorch for some operators (like advanced indexing).

But let's keep the behavior similar to what detectron2 does for now.